rust/src/router.rs

```diff
@@ -1083,7 +1097,9 @@ impl Router {
     ) -> Result<PipeProducerToRouterPair, PipeProducerToRouterError> {
         debug!("pipe_producer_to_router()");
 
-        if pipe_to_router_options.router.id() == self.id() {
+        let PipeToRouterOptions { keep_id, .. } = pipe_to_router_options;
+
+        if keep_id && pipe_to_router_options.router.id() == self.id() {
```

Comment:
This is inconsistent. Let's either do this:
```suggestion
        if pipe_to_router_options.keep_id && pipe_to_router_options.router.id() == self.id() {
```
Or if you want to make the line shorter then this:
```suggestion
        let PipeToRouterOptions { router, keep_id, .. } = &pipe_to_router_options;

        if *keep_id && router.id() == self.id() {
```

Reply:
We **need** to extract `keep_id` **before** moving `pipe_to_router_options` to `self.get_or_create_pipe_transport_pair()` so your proposal doesn't compile.

Reply:
I don't understand what doesn't compile. Which option have you tried and what was the compiler error?

Reply:
I've done it here: https://github.com/versatica/mediasoup/pull/1604/commits/d736b90ff67a7fbb45a13114fa80d48483c4b124

Reply:
I never liked `ref`, but sure, that works too.